mw-webapp/src/hooks/useErrorHandler.ts

```diff
@@ -0,0 +1,42 @@
+import {useEffect, useState} from "react";
+
+/**
+ * Custom hook to handle global unhandled errors and promise rejections, returns error message
+ */
+export const useErrorHandler = () => {
+  const [error, setError] = useState<string>("");
+
+  /**
+   * Sets a new error message, ensuring the state updates even if the message is the same
+   */
+  const setNewError = (newError: string) => {
+    setError("");
+    setTimeout(() => setError(newError), 0);
```

Comment:
probably we can handle it different way. Lets save an object in state instead of string. in that case solution will be scalabel (you will be able add more fields in future) and you will get rid of this workaround with resetting and resetting the state

Reply:
Problem is, Notification will not be re-rendered with this solution because message is the same. Only working solution that i found(except of resetting the state) is setting key Date.now() to Notification component so it will re-render